core/src/main/scala/sttp/tapir/server/interpreter/EncodeOutputs.scala

```diff
@@ -59,18 +62,39 @@ class EncodeOutputs[B, S](rawToResponseBody: ToResponseBody[B, S]) {
         )
       case EndpointOutput.OneOf(mappings, _) =>
         val enc = encoded[Any]
-        val mapping = mappings
-          .find(mapping => mapping.appliesTo(enc))
-          .getOrElse(throw new IllegalArgumentException(s"No status code mapping for value: $enc, in output: $output"))
-        apply(mapping.output, ParamsAsAny(enc), mapping.statusCode.map(ov.withStatusCode).getOrElse(ov))
+
+        val bodyMappings: Map[MediaType, StatusMapping[_]] = mappings
+          .filter(_.appliesTo(enc))
+          .collect({
+            case sm @ StatusMapping(_, EndpointIO.Body(bodyType, codec, _), _) =>
```

Comment:
the body outputs might be nested, e.g. `.output(jsonBody[X].map(...))` or `.output(jsonBody[X].and(header(...)))`, so such a simple pattern match won't work in this case. I think you'll have to traverse the outputs to find the body

Reply:
yes, I keep forgetting about this nest here, adjusted